File: app/src/main/java/forpdateam/ru/forpda/common/receivers/WakeUpReceiver.java
package forpdateam.ru.forpda.common.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import forpdateam.ru.forpda.notifications.NotificationsService;

/**
 * Created by isanechek on 7/11/17.
 */

public class WakeUpReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        Log.d("SUKA", "RECIEVER ACTION " + intent.getAction());
        String action = intent.getAction();
        if (action != null) {
            if (action.equals(Intent.ACTION_SCREEN_ON)) {
                NotificationsService.startAndCheck();
            } else if (action.equals(Intent.ACTION_BOOT_COMPLETED)) {
                NotificationsService.startAndCheck();
            }
        }
    }
}

File: app/src/main/java/forpdateam/ru/forpda/common/receivers/WakeUpReceiver.kt
package forpdateam.ru.forpda.common.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import forpdateam.ru.forpda.model.interactors.events.EventsController
import forpdateam.ru.forpda.notifications.NotificationsService
import ru.radiationx.quill.getScope

/**
 * Created by isanechek on 7/11/17.
 */
class WakeUpReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        Log.d("SUKA", "RECIEVER ACTION " + intent.action)
        context.getScope().get(EventsController::class).start()
        NotificationsService.startAndCheck(context)
    }
}
